### PR TITLE
HHH-14023: Adjust H2Dialect for 1.4.201: sequence handling, DECIMAL, NUMERIC and VARBINARY

### PR DESCRIPTION
**Pull request by famod**
See also:
- https://github.com/h2database/h2database/issues/2625
- https://hibernate.zulipchat.com/#narrow/stream/132096-hibernate-user/topic/Locale.20problem.20when.20building.20hibernate-orm

cc @Sanne

To test this, you will need a current h2 SNAPSHOT (build it yourself from `master`) _with adjusted `org.h2.engine.Constants.BUILD_ID`_ (which is still on 200 until the release of 201).
I was able to get my local SNAPSHOT in to Gradle via:
```diff
diff --git a/gradle/java-module.gradle b/gradle/java-module.gradle
index 1928f00d42..8d2ea3dbbc 100644
--- a/gradle/java-module.gradle
+++ b/gradle/java-module.gradle
@@ -5,6 +5,11 @@
  * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
  */

+repositories {
+       mavenLocal()
+       mavenCentral()
+}
+
 buildscript {
        repositories {
                mavenCentral()
@@ -81,7 +86,7 @@ dependencies {
        testRuntime( libraries.byteBuddy )

        //Databases
-       testRuntime( libraries.h2 )
+       testRuntime( libraries.h2 ) { changing = true }
        testRuntime( libraries.hsqldb )
        testRuntime( libraries.postgresql )
        testRuntime( libraries.mysql )
diff --git a/gradle/libraries.gradle b/gradle/libraries.gradle
index cc53eb9448..8481c51267 100644
--- a/gradle/libraries.gradle
+++ b/gradle/libraries.gradle
@@ -10,7 +10,7 @@
 ext {

     junitVersion = '4.12'
-    h2Version = '1.4.196'
+    h2Version = '1.4.201-SNAPSHOT'
     bytemanVersion = '4.0.8' //Compatible with JDK14
     jnpVersion = '5.0.6.CR1'

@@ -177,6 +177,9 @@ ext {
 }

 configurations.all {
+    resolutionStrategy {
+      cacheChangingModulesFor 0, 'seconds'
+    }
     resolutionStrategy.eachDependency { DependencyResolveDetails details ->
         //Force the "byte buddy agent" version to match the Byte Buddy version we use, as Mockito might p
         if (details.requested.group + ":" + details.requested.name == 'net.bytebuddy:byte-buddy-agent') {
```
⚠️ You will see _many_ failures with this but _without_ this PR you would see even more errors.

### CODE REVIEW

**Comment by Sanne** [comment]
Hi, looks good so far :+1:  Please notify me again if you make progress.

Regarding  d47e44e : if you send it as a separate PR and JIRA, we can merge that already.

**Comment by famod** [comment]
Well, I thought we could merge this entire PR?
My plan was to create more PRs for the other, older versions but that will take time.

**Comment by Sanne** [comment]
ok, yes I suppose we could entertain the idea of merging this as is since it doesn't seem to break anything.. but it doesn't fix anything either, so don't you think the JIRA is going to look a bit confusing to people?

We could word it differently, but since H2 1.4.201 hasn't been released we can't claim much about compatibility with it either.

**Comment by famod** [comment]
> but it doesn't fix anything either

Well, it does fix the sequence handling (https://github.com/h2database/h2database/issues/2625#issuecomment-629667698) and the three datatypes (https://github.com/h2database/h2database/issues/2625#issuecomment-629734255) when using Hibernate with 1.4.201-SNAPSHOT (and I guess also with the final 1.4.201).
_Without_ this PR there are more Hibernate testsuite failures than _with_ this PR when running the testsuite with h2 1.4.201-SNAPSHOT.

How about rewording to "HHH-14023 Adjust H2Dialect for 1.4.201: sequence handling, DECIMAL, NUMERIC and VARBINARY"

**Comment by Sanne** [comment]
> Without this PR there are more Hibernate testsuite failures than with this PR when running the testsuite with h2 1.4.201-SNAPSHOT.

Sure, it's progress and I wouldn't want to lose it. I just want to make sure that people don't get the wrong expectation of assuming it being fully compatible.

+1 for the wording alternatives you proposed. Ideally if you could split up the language parameter to a different ticket so to allow independent forward/back porting that would be great.

**Comment by famod** [comment]
@Sanne I've moved the language change to #3409 and I also renamed this PR. Should be good to merge now, no?

**Comment by famod** [comment]
@Sanne Can we get this merged? IMO this should not be blocked by my build result troubles in #3412.

**Comment by Sanne** [comment]
hi @famod ! sure, I have no objections. Just a bit busy - I'll try to have enother look at this tomorrow

**Comment by dreab8** [comment]
Aplied upstream,

Thanks @famod
